lib/awful/widget/watch.lua

```diff
@@ -0,0 +1,62 @@
+---------------------------------------------------------------------------
+--- Watch widget.
+--
+-- @release @AWESOME_VERSION@
+-- @classmod awful.widget.watch
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local textbox = require("wibox.widget.textbox")
+local timer = require("gears.timer")
+local spawn = require("awful.spawn")
+
+local watch = { mt = {} }
+
+--- Create a textbox that shows the output of a command
+-- and updates it at a given time interval.
+--
+-- @tparam string command The command.
+--
+-- @tparam int timeout (optional) The time interval at which the textbox
+-- will be updated. Default to 5.
+--
+-- @tab callback Function (optional) The function that will be applied to
+-- the output before it is shown in the textbox.
+-- @param callback.widget Base widget instance.
+-- @tparam string callback.stdout Output on stdout.
+-- @tparam string callback.stderr Output on stderr.
+-- @tparam string callback.exitreason Exit Reason.
+-- The reason can be "exit" or "signal".
+-- @tparam integer callback.exitcode Exit code.
+-- For "exit" reason it's the exit code.
+-- For "signal" reason — the signal causing process termination.
+--
+-- @param base_widget (optional) Defaults to `wibox.widget.textbox()`
+--
+-- @return The widget used by this watch
+function watch.new(command, timeout, callback, base_widget)
+    timeout = timeout or 5
+    base_widget = base_widget or textbox()
+    callback = callback or function(widget, stdout, stderr, exitreason, exitcode) -- luacheck: no unused args
+        widget:set_text(stdout) -- Your stdout or stderr does not work, the empty string is "trueish"
```

Comment:
Heh, sorry for the confusion. I didn't mean you to add that comment here, you can remove that. What I wanted to say was that your old version of this code doesn't work.

You did: `set_text(stdout or stderr)` (or something like that). However, `stdout` will never be `false` or `nil` and thus this always ignored `stderr` besides the code looking otherwise. The comment was meant to mention this to you, not something that I wanted you to copy into this PR.

Again, sorry for the confusion.

(Oh and thanks a lot for running this against Luacheck, that makes my attempts to automatically run Luacheck on Travis easier)